xdp-filter/test-xdp-filter.sh

```diff
@@ -261,6 +261,38 @@ test_print()
     check_run $XDP_FILTER unload $NS -v
 }
 
+run_python_test()
+{
+    python3 -c "import xdp_test_harness; import bcc;"
```

Comment:
See the other comment. For now, please just check that python itself exists (and it may be just 'python' not 'python3' - so maybe a detection and support for a $PYTHON variable is in order?)

Reply:
> Did not look at all the details, but I think we need to update the pre-run check to make sure we have python3 and the required python modules (scapy/etc.).

This line here returns non-zero if one of the required libraries is not installed. I am checking only for XDP test harness and BCC, since both Scapy and Pyroute2 are listed as dependencies for XDP test harness and so should be also installed by the package manager.
This should also check whether Python exists, since if that is the case the shell will return non-zero value. Python 2 will not work, because the library uses some features of Python 3.

Reply:
Yup, you should definitely check that python itself exists as well. The version check will be taken care of by the import check, I think (the module won't import correctly in py2). However, python3 can be called just 'python' on some systems; and we may have build systems where it's set up weirdly, so supporting a $PYTHON variable seems sensible...

Reply:
Note that these tests can also be run from the Github clone, so I think checks for scapy and pyroute2 should also be added and any other dependency.

Reply:
Hmm, I was assuming that was implicit in 'import xdp_test_harness' as that would in turn try to import scapy and pyroute2. But not sure that's actually the case for just this import? If not, I agree, we should check for the other modules as well...

Reply:
Using `import xdp_test_harness` does not check for Scapy or pyroute2. What I meant was that pip while installing XDP test harness will also install its dependencies: Scapy and pyroute2, meaning if XDP test harness is installed then so should be the dependencies. But I am okay with adding the libraries explicitly into the check.

Reply:
Hmm, right. Well I think it's better to make sure the dependencies are also there, but that could also be done by just importing them from the top-level xdp_test_harness module. That would make sure we check for broken installations while still keeping the dependencies in the module itself, not in this script...